jdbc/src/main/java/org/apache/zeppelin/jdbc/kyuubi/KyuubiUtils.java

```diff
@@ -0,0 +1,118 @@
+
+import org.apache.commons.dbcp2.DelegatingConnection;
+import org.apache.commons.dbcp2.DelegatingStatement;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.kyuubi.jdbc.hive.KyuubiConnection;
+import org.apache.kyuubi.jdbc.hive.KyuubiStatement;
+import org.apache.zeppelin.interpreter.InterpreterContext;
+import org.apache.zeppelin.jdbc.JDBCInterpreter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.sql.Statement;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class include hive specific stuff.
+ * e.g. Display hive job execution info.
+ *
+ */
+public class KyuubiUtils {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(KyuubiUtils.class);
+  private static final int DEFAULT_QUERY_PROGRESS_INTERVAL = 1000;
+  /**
+   * Display hive job execution info
+   */
+  public static void startMonitorThread(Connection conn,
+                                        Statement stmt,
+                                        InterpreterContext context,
+                                        boolean displayLog,
+                                        JDBCInterpreter jdbcInterpreter) {
+    KyuubiConnection kyuubiConn = (KyuubiConnection)
+            ((DelegatingConnection<?>) ((DelegatingConnection<?>) conn).getDelegate()).getDelegate();
+    KyuubiStatement kyuubiStmt = (KyuubiStatement)
+            ((DelegatingStatement) ((DelegatingStatement) stmt).getDelegate()).getDelegate();
+    // need to use final variable progressBar in thread, so need progressBarTemp here.
+    final ProgressBar progressBar = new ProgressBar();
+    final long queryInterval = Long.parseLong(
+            jdbcInterpreter.getProperty("zeppelin.jdbc.kyuubi.monitor.query_interval",
+                    DEFAULT_QUERY_PROGRESS_INTERVAL + ""));
+    Thread thread = new Thread(() -> {
+      String jobUrlTemplate = jdbcInterpreter.getProperty("zeppelin.jdbc.kyuubi.jobUrl.template");
+      boolean jobUrlExtracted = false;
+
+      try {
+        while (kyuubiStmt.hasMoreLogs() && !kyuubiStmt.isClosed() && !Thread.interrupted()) {
+          Thread.sleep(queryInterval);
+          List<String> logs = kyuubiStmt.getExecLog();
+          String logsOutput = StringUtils.join(logs, System.lineSeparator());
+          LOGGER.debug("Kyuubi job output: {}", logsOutput);
+          boolean displayLogProperty = context.getBooleanLocalProperty("displayLog", displayLog);
+          if (displayLogProperty && !StringUtils.isBlank(logsOutput)) {
+            context.out.write(logsOutput + "\n");
+            context.out.flush();
+            progressBar.operationLogShowedToUser();
+          }
+
+          if (!jobUrlExtracted) {
+            String appId = kyuubiConn.getEngineId();
+            String appUrl = kyuubiConn.getEngineUrl();
+            String jobUrl = null;
+            // prefer to use customized template, and fallback to Kyuubi returned engine url
```

Comment:
in case the Hadoop cluster is deployed behind Gateway, e.g. Knox, the users at the corp zone may need to use a different URL to access the Application Web page